### PR TITLE
update wombat integration

### PR DESCRIPTION
**Pull request by koala0814**
1. update pricing for high cov ratio pool and dynamic pool
2. implement BMW event listener for pool fetching
3. implement wombat pool based on StatefulRpcPoller

### CODE REVIEW

File: src/dex/wombat/wombat-integration.test.ts [hunk 1]
```diff
@@ -207,7 +231,7 @@ describe('Wombat', function () {
         destTokenSymbol,
         SwapSide.SELL,
         amountsForSell,
-        '', // TODO: Put here proper function name to check pricing
+        'quotePotentialSwap', // TODO: Put here proper function name to check pricing
```
Comment (jack-wombat) [inline src/dex/wombat/wombat-integration.test.ts:234]: nit: do we still need the TODO comment?

File: src/dex/wombat/wombat-integration.test.ts [hunk 2]
```diff
@@ -221,7 +245,7 @@ describe('Wombat', function () {
         destTokenSymbol,
         SwapSide.BUY,
         amountsForBuy,
-        '', // TODO: Put here proper function name to check pricing
+        'quotePotentialSwap', // TODO: Put here proper function name to check pricing
```
Comment (jack-wombat) [inline src/dex/wombat/wombat-integration.test.ts:248]: nit: do we still need the TODO comment?

**Review by jack-wombat — APPROVED** [review]
LGTM high level. Is there support for dynamic pool?

File: src/dex/wombat/wombat-integration.test.ts (hunk 1 above)
Comment (koala0814) [reply src/dex/wombat/wombat-integration.test.ts]: removed

File: src/dex/wombat/wombat-bmw.ts [hunk 3]
```diff
@@ -107,37 +110,68 @@ export class WombatBmw extends StatefulEventSubscriber<BmwState> {
...
+      });
+    });
+
+    returnData = await this.dexHelper.multiContract.methods
+      .tryAggregate(false, inputs)
+      .call({}, blockNumber);
+    i = 0;
+    pool2AssetInfo.forEach((asset2TokenMap, pool) => {
+      // don't add cross-chain pools
+      if (!returnData[i++].success) {
```
Comment (jack-wombat) [inline src/dex/wombat/wombat-bmw.ts:170]: Q: why not add cross chain pool? Some chain like avax only has cross chain pools.

**Review by jack-wombat — APPROVED** [review]
LGTM

File: src/dex/wombat/wombat-bmw.ts (hunk 3 above)
Comment (koala0814) [reply src/dex/wombat/wombat-bmw.ts]: Fixed. Cross-chain pool is added back now.

**Pull request merged**